drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
> For the crash console, wouldn't it be possible for the coreboot console to allocate some scratch space for its plat_crash_console_putc() implementation to use instead of registers? Any initialization needed can be done within plat_crash_console_init().

Well... I could have each function store its return pointer in static scratch space, yeah. Since the crash console is not reentrant that should work. It would just look kinda weird.

But going back to your idea with the "splitting" driver, I also don't see how I could do that without running into naming clashes. The generic console code just calls function names like console_core_putc, and the individual drivers implement those. I can't really shim a layer in-between that and leave both of the existing sides untouched (at least not without heavy linker magic). I think this only really works if the splitting happens in the generic code already (or if the whole console API was made more flexible and would work by registering function pointers at runtime or something like that).

> The general philosophy in TF is that platform code is precisely where variations like this should be implemented. We want to work towards reducing build flags in generic code, not proliferating them.

Does that also count for flags which represent user decisions rather than modeling hardware? As I tried to say, I don't think console types (the ones you'd like to mix and match) are really a property of the platform. Sure, some types may only be supported on some subset of platforms, but the decision on which of them to enable for a particular build is made by the person building it, not predetermined by the hardware. So I feel like this is similar to something like DEBUG=1 or LOG_LEVEL=30, not a hardware-describing option like ARM_CCI_PRODUCT_ID=500. For those, doesn't it make much more sense to keep them in generic code so they don't need to be duplicated individually by all platforms?